lms/static/js/demographics_collection/DemographicsCollectionBanner.jsx

```diff
@@ -8,47 +9,78 @@ export class DemographicsCollectionBanner extends React.Component {
     super(props);
     this.state = {
       modalOpen: false,
+      hideBanner: false
+    }
+
+    this.dismissBanner = this.dismissBanner.bind(this);
+  }
+
+  async dismissBanner(e) {
+    // The dismiss button is nested inside an <a>, so we need to call stopPropogation here to
+    // keep the Modal from also opening when the Dismiss button is clicked
+    e.stopPropagation();
+
+    const requestOptions = {
+      method: 'PATCH',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': Cookies.get('csrftoken'),
+      },
+      body: JSON.stringify({
+        show_call_to_action: false,
+      })
+    };
+
+    let response = {};
+    response = await fetch(`${this.props.lmsRootUrl}/api/demographics/v1/demographics/status/`, requestOptions);
+    if (response.status === 200) {
+      this.setState({ hideBanner: true });
     }
   }
 
   render() {
-    return (
-      <div>
-        <a id="demographics-banner-link" className="btn" onClick={() => this.setState({ modalOpen: true })}>
-          <div
-            className="demographics-banner d-flex justify-content-lg-between flex-row py-1 px-2 mb-2 mb-lg-4"
-            role="dialog"
-            aria-modal="false"
-            aria-label="demographics questionnaire pitch"
-          >
-            <div className="d-flex justify-content-left align-items-lg-center flex-column flex-lg-row  w-100">
-              <img className="demographics-banner-icon d-none d-lg-inline-block" src={this.props.bannerLogo} alt="" aria-hidden="true" />
-              <div className="demographics-banner-prompt d-inline-block font-weight-bold text-white mr-4 py-3 px-2 px-lg-3">
-                {gettext('Want to make edX better for everyone?')}
+    if (!(this.state.hideBanner)) {
```

Comment:
Is this mostly just shifting the code for the conditional?

Reply:
Mostly, The only notable change in the react component is that I added an onClick to line #65 that calls the `dismissBanner` function.